examples/java-matter-controller/java/src/com/matter/controller/commands/common/Command.kt

```diff
@@ -0,0 +1,256 @@
+    addArgumentToList(arg)
+  }
+
+  /**
+   * @brief Add a long Integer command argument
+   * @param name The name that will be displayed in the command help
+   * @param min The minimum value of the argv value
+   * @param max The minimum value of the argv value
+   * @param out A pointer to a MutableInteger where the argv value will be stored
+   * @param desc The description of the argument that will be displayed in the command help
+   * @param optional Indicate if an optional argument
+   * @return The number of arguments currently added to the command
+   */
+  fun addArgument(
+    name: String?, min: Short, max: Short, out: AtomicLong?, desc: String?, optional: Boolean
+  ) {
+    val arg = Argument(name!!, min, max, out!!, desc, optional)
+    addArgumentToList(arg)
+  }
+
+  /**
+   * @brief Add a long Integer command argument
+   * @param name The name that will be displayed in the command help
+   * @param min The minimum value of the argv value
+   * @param max The minimum value of the argv value
+   * @param out A pointer to a AtomicLong where the argv value will be stored
+   * @param desc The description of the argument that will be displayed in the command help
+   * @param optional Indicate if an optional argument
+   * @return The number of arguments currently added to the command
+   */
+  fun addArgument(
+    name: String?, min: Long, max: Long, out: AtomicLong?, desc: String?, optional: Boolean
+  ) {
+    val arg = Argument(name!!, min, max, out!!, desc, optional)
+    addArgumentToList(arg)
+  }
+
+  /**
+   * @brief Add an IP address command argument
+   * @param name The name that will be displayed in the command help
+   * @param out A pointer to a IPAddress where the argv value will be stored
+   * @param optional Indicate if an optional argument
+   * @return The number of arguments currently added to the command
+   */
+  fun addArgument(name: String?, out: IPAddress?, optional: Boolean) {
+    val arg = Argument(name!!, out!!, optional)
+    addArgumentToList(arg)
+  }
+
+  /**
+   * @brief Add a String command argument
+   * @param name The name that will be displayed in the command help
+   * @param out A pointer to a StringBuffer where the argv value will be stored
+   * @param desc The description of the argument that will be displayed in the command help
+   * @param optional Indicate if an optional argument
+   * @return The number of arguments currently added to the command
+   */
+  fun addArgument(
+    name: String?, out: StringBuffer?, desc: String?, optional: Boolean
+  ) {
+    val arg = Argument(name!!, out!!, desc, optional)
+    addArgumentToList(arg)
+  }
+
+  /**
+   * @brief Initialize command arguments
+   * @param argc The number of arguments to a command (Does not include command itself)
+   * @param args Supplied command-line arguments as an array of String objects.
+   */
+  fun initArguments(argc: Int, args: Array<String>) {
```

Comment:
Can you use the size of `args` instead of passing in `argc` as well? It might be slightly simpler that way.